packages/client/hmi-client/src/workflow/ops/model-from-code/tera-model-from-code.vue

```diff
@@ -349,16 +349,57 @@ function removeCodeBlock(index: number) {
 }
 
 function isModelValid() {
-	if (selectedModelFramework.value === ModelFramework.Petrinet) {
+	if (clonedState.value.modelFramework === ModelFramework.Petrinet) {
 		return !!selectedModel.value;
 	}
 
-	if (selectedModelFramework.value === ModelFramework.Decapodes) {
+	if (clonedState.value.modelFramework === ModelFramework.Decapodes) {
 		return decapodesModelValid.value;
 	}
 
 	return false;
 }
+
+async function fetchModel() {
+	if (!clonedState.value.modelId) {
+		selectedModel.value = null;
+		return;
+	}
+	isProcessing.value = true;
+	const model = await getModel(clonedState.value.modelId);
+	selectedModel.value = model;
+	isProcessing.value = false;
+}
+
+// watch for model id changes on state
+watch(
+	() => clonedState.value.modelId,
+	async () => {
+		await fetchModel();
+	}
+);
+
+// watch for prop state changes
+watch(
+	() => props.node.state,
+	() => {
+		clonedState.value = cloneDeep(props.node.state);
+	}
+);
+
+watch(
+	() => props.node.active,
+	async () => {
+		if (props.node.active) {
+			selectedOutputId.value = props.node.active;
+		}
+	}
+);
```

Comment:
You have quite a few watchers here, are we confident that they are mutually exclusive and will not interfere with each other?

Reply:
I was a little sad at the watchers, but I couldn't really find a way to get around them here since we need to listen to the prop changes.  In any case I believe that the watchers shouldn't be interfering with one another.

Reply:
Why do you need to watch for those node information?

Reply:
I was able to make the selectedOutputId a computed property instead of a watcher ✅ .  I believe I need to watch the state however.  Since the drilldown itself has the ref of `clonedState`, we need to listen whenever the output is changed which in turn changes the state and copies to `clonedState`.